frontend/rust-lib/flowy-database2/src/manager.rs

```diff
@@ -194,12 +194,18 @@ impl DatabaseManager2 {
       |user_database| {
         let mut params = CreateViewParams::new(database_id.clone(), database_view_id, name, layout);
         if let Some(database) = user_database.get_database(&database_id) {
-          if let Some((field, layout_setting)) = DatabaseLayoutDepsResolver::new(database, layout)
+          match DatabaseLayoutDepsResolver::new(database, layout)
             .resolve_deps_when_create_database_linked_view()
           {
-            params = params
-              .with_deps_fields(vec![field])
-              .with_layout_setting(layout_setting);
+            Some((Some(field), layout_setting)) => {
+              params = params
+                .with_deps_fields(vec![field])
+                .with_layout_setting(layout_setting);
+            },
+            Some((None, layout_setting)) => {
+              params = params.with_layout_setting(layout_setting);
+            },
+            _ => {},
```

Comment:
Suggestion:

```rs
if let Some(field) = field {
  params = params.with_deps_fields(vec![field]);
}
params = params.with_layout_setting(layout_setting);
```

No need to match again.

Reply:
Got it, thanks!